### PR TITLE
ENG-2486 Announcement Banner to announce new releases

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR adds a small UI at the top of the screen to alert users that there is a new release of Aqueduct.

It also has logic to check local storage to see if the user has dismissed the banner, or if the banner being shown is for a newer version than the user has last seen on their browser.

## Related issue number (if any)
ENG-2486

## Loom demo (if any)
https://www.loom.com/share/761c534ad6804b90a2c8c1f5d8a9e2f6

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/layouts/default.tsx [hunk 1]
```diff
@@ -38,9 +39,14 @@ export const DefaultLayout: React.FC<Props> = ({
   const muiTheme = createTheme(theme);
   const dispatch: AppDispatch = useDispatch();
 
+  const [showBanner, setShowBanner] = useState<boolean>(false);
+
   useEffect(() => {
     if (user) {
       dispatch(handleFetchNotifications({ user }));
+      // TODO: Get the version number here.
```
Comment (agiron123) [inline src/ui/common/src/components/layouts/default.tsx:47]: remove comments, this is no longer needed.

File: src/ui/common/src/components/layouts/menuSidebar.tsx [hunk 2]
```diff
@@ -140,6 +141,10 @@ const MenuSidebar: React.FC<{
     fetchVersionNumber();
   }, [user.apiKey]);
 
+  const { data, error, isLoading } = useVersionNumberGetQuery({
```
Comment (agiron123) [inline src/ui/common/src/components/layouts/menuSidebar.tsx:144]: Remove, this didn't quite work out for some reason, need to dig through the RTK docs some more and learn about how to send request with no params, and just the API key header.

**Comment by vsreekanti** [comment]
@agiron123, nice work! A few things: 
1. The banner looks quite large to me. Could we make it smaller? 
2. The color of the banner looks a little out of place to me. We're not really using that seafoam green anywhere else in the UI. Can we try either a light blue like the logo light blue or light grey (my preference)? 
3. Can we add a flag to not always show the banner? I find it kinda distracting on the DAGs page, so I'd prefer to only show it on the home, workflows list, integrations list, and data list pages.

**Comment by agiron123** [comment]
> @agiron123, nice work! A few things:
> 
> 1. The banner looks quite large to me. Could we make it smaller?
> 2. The color of the banner looks a little out of place to me. We're not really using that seafoam green anywhere else in the UI. Can we try either a light blue like the logo light blue or light grey (my preference)?
> 3. Can we add a flag to not always show the banner? I find it kinda distracting on the DAGs page, so I'd prefer to only show it on the home, workflows list, integrations list, and data list pages.

@vsreekanti 
1. I think the banner is a bit large as well, I made it 64px so that it matches the size of the logo. Originally, I was hoping to put the banner over both the sidebar and navbar, but I was having some issues where the bottom footer of the Sidebar didn't render correctly. I'll post a screenshot here of a smaller banner.
2. Light grey sounds good with me.
3. Sure, I can add flags on which pages to show the banner.

**Comment by agiron123** [comment]
@vsreekanti here is a screenshot of the banner smaller and with gay background. I also took the text size down as well:
image

**Comment by agiron123** [comment]
@vsreekanti Updated demo can be found here:
https://www.loom.com/share/1d038acc86234b7bb9b989272799d342

**Comment by vsreekanti** [comment]
Yep, this it awesome — thanks Andre! 

Can we update the text to say: 

> Aqueduct v{version_number} is out! See release notes →

**Review by vsreekanti — APPROVED** [review]

**Comment by agiron123** [comment]
> Yep, this it awesome — thanks Andre!
> 
> Can we update the text to say:
> 
> > Aqueduct v{version_number} is out! See release notes →

@vsreekanti sure thing!

**Pull request merged**